src/Illuminate/Http/Client/RequestException.php

```diff
@@ -21,6 +21,8 @@ class RequestException extends Exception
      */
     public function __construct(Response $response)
     {
+        parent::__construct("{$response->effectiveUri()} returned status code {$response->status()}", $response->status());
```

Comment:
Setting the response status as the exception code is unwanted. HTTP error codes aren't exception error codes.

Reply:
Looks like Guzzle set the response status code as the exception code https://github.com/guzzle/guzzle/blob/cce7ba5b8e2ea61f21f7101b853edb3c783dd3e1/src/Exception/RequestException.php#L30-L31, since the Laravel HTTP client is a wrapper around Guzzle wouldn't it be better to keep this behavior ?

Reply:
Seems like you're correct 👍